### PR TITLE
Onboarding Improvements: New Login Epilogue Screen Design

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent #14990

This PR adds a couple of new designs to the login epilogue screen, the `normal` and `expanded` ones.
- The `normal` design is the one that will be shown when the number of sites equal to 2-3 or less.
- The `expanded` design is the one that will be shown when the number of sites end up being more than 3.

Sites | Existing | New
-----|-----|-----
2-3 or less<br/>(`normal`) | existing_normal | new_normal
More than 3<br/>(`expanded`) | existing_expanded | new_expanded

-----

To test:

#### Scenario: `normal`
1. Find an account which has lots of sites and log-in to the app.
2. Go to the `Side Chooser` screen and hide all but 3 of the existing sites. You should at max have no more than 3 sites shown.
2. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
3. Enable the `OnboardingImprovementsFeatureConfig` feature flag.
4. Log-out and log-in to the app again.
5. Make sure the new `Login Epilogue` screen looks as expected, the `normal` design is shown.

#### Scenario: `expanded`
1. Find an account which has lots of sites and log-in to the app.
2. Go to the `Side Chooser` screen and unhide all sites. You should at least have more that 3 sites shown.
2. Go to `My Site` -> `App Settings` -> `Test feature configuration`.
3. Enable the `OnboardingImprovementsFeatureConfig` feature flag.
4. Log-out and log-in to the app again.
5. Make sure the new `Login Epilogue` screen looks as expected, the `expanded` design is shown.

-----

#### Merge Instructions:
- ~~@osullivanchris to do a design review of both the new `Login Epilogue` screens (`normal` and `expanded`).~~
- ~~Remove the `[Status] Needs Design Review`.~~
- ~~Merge [WordPress-Login-Flow-Android#60](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/60).~~
- ~~Update `Login` hash from the PR (`60-{sha1}`) to `develop-{sha1}`.~~
- Merge this PR.

-----

#### Notes

This PR is UI related only, the below functionality will be handled in separate PRs:
- Selecting a `Site`, which will either show the `Quick Start Prompt` or navigate to `My Site` directly.
- Clicking the `Create a new site` button, which will start the `WordPress.com` site creation flow.
- `Landscape` UI, which requires a bit more fine tuning before being ready.

-----

## Regression Notes
1. Potential unintended areas of impact

The existing `Login Epilogue` screen might be somehow affected by all the changes in this PR since all 3 screen are sharing all the logic, that is code-wise.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by osullivanchris — CHANGES_REQUESTED** [review]
Thanks for sharing @ParaskP7 , its looking great already. Here are a few things I noticed:

- the dividing line is going straight through the 'or' text, rather than breaking either side
- in landscape mode, with only one site, the 'create site' button is not sticky and is off screen for me. I think it should always be sticky in landscape mode for both 1-3 sites and more site versions.
- the buttons are not wired up yet. I couldn't tap a site. 'Create new site' took me to My Site. Assume this is just not done yet.
- Seeing the 'Choose a site to open' text in serif, I'm not sure we've got the balance right with type sizes yet. Let's discuss options together next week.

File: WordPress/src/main/res/layout/login_epilogue_screen_new.xml
Comment (ashiagr) [inline WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: Suggestion 💡: Since this `orLayout` is now used at three places:

`login_email_screen.xml`
`login_epilogue_screen_new.xml`
`login_epilogue_screen_new_expanded.xml`

how about it is extracted to a separate layout and included in all three places? This way, we wouldn't have to worry about the dividing line going through the "or" text as noticed by Chris in the future.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java:114]: Warning ⚠️:  IIRC we discussed not making any changes to the epilogue screen for the Jetpack app or at least not show the create site option. I'd suggest either we update the condition for the appropriate layout display in this PR or add a note in the task to not forget about it.

**Review by ashiagr — COMMENTED** [review]
Great job @ParaskP7! Changes are coming along really nicely  🙌. Also, appreciate the in-between refactoring you did to make the code more readable and robust. 

I only left one suggestion and a warning for you to consider. Hope it helps.

**Comment by ParaskP7** [comment]
👋 @osullivanchris !

Thanks for much for this detailed review! 🙏 

> - the dividing line is going straight through the 'or' text, rather than breaking either side

You are totally right, really good eye, I totally missed that. It is very strange that this is happening since I copy-pasted the component from the `Google` sign-in form as it was specified there, thus I never even bother to notice it afterwards. Hmmm... I wonder... I'll update it asap and let you know! 👍 

> - in landscape mode, with only one site, the 'create site' button is not sticky and is off screen for me. I think it should always be sticky in landscape mode for both 1-3 sites and more site versions.
> - the buttons are not wired up yet. I couldn't tap a site. 'Create new site' took me to My Site. Assume this is just not done yet.

I have left a note on my description on all that. `Selecting a Site`, `Clicking the Create a new site` button and the `Landscape UI` will be done in individual PR, separate to this one, you can also check the list of todos in the parent issue, [here](https://github.com/wordpress-mobile/WordPress-Android/issues/14990). Also, I just added the `Jetpack App` todo item, which will be done in a separate PR as well. Divide and conquer! 😄 

> - Seeing the 'Choose a site to open' text in serif, I'm not sure we've got the balance right with type sizes yet. Let's discuss options together next week.

Definitely, let's discuss our options together. In this instance I used the `@style/Widget.LoginFlow.Button.Secondary` style that is already the style for the secondary (`white`) button during the login flow, as it is already defined by the library. I didn't want to make any changes to that before you can see the result and suggest updates to the button. 👍

File: WordPress/src/main/res/layout/login_epilogue_screen_new.xml
Comment (ParaskP7) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: 👋 @ashiagr !

Such a nice suggestion, totally, I agree 💯 and will do that asap!

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java
Comment (ParaskP7) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java:114]: 👋 @ashiagr !

Great observation, I added a additional todo task on the parent issue [here](https://github.com/wordpress-mobile/WordPress-Android/issues/14990), which will deal with everything `Jetpack App` related. I think it is better for us to dedicate some additional time, add more thought and testing to when it comes to the Jetpack app. This way, it will also make this PR less complex and have it focusing on the WordPress app alone, at least until the full solution is complete. Wdyt?

Comment (ashiagr) [reply WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginEpilogueFragment.java:114]: Works for me 🙂, thanks for adding the additional todo task 🙏.

File: WordPress/src/main/res/layout/login_epilogue_screen_new.xml
Comment (ParaskP7) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: 👋 @ashiagr !

I have created this [WordPress-Login-Flow-Android#60](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/60) PR which extract the `login-or-layout` as discussed. Let me know your thoughts. Then, I'll incorporate this change within this PR.

PS: @osullivanchris , this [WordPress-Login-Flow-Android#60](https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/60) PR above slightly changes the UI on the `login-email-screen` for just a `2dp` bottom diff, between the `Or Layout` and the `Google Sign-In` button. This change is barely noticeable but I chose to make it to make this `Or Layout` consistent across all screens afterwards. Are you okay with this change? PS: When this change gets incorporated in this PR I can ping you again so that you can double check the `login-email-screen` and see if you don't like something, maybe those extra `2dp` will create an issue of some kind, design wise.

**Comment by osullivanchris** [comment]
Thanks for the reply @ParaskP7 

> You are totally right, really good eye, I totally missed that. It is very strange that this is happening since I copy-pasted the component from the Google sign-in form as it was specified there, thus I never even bother to notice it afterwards. Hmmm... I wonder... I'll update it asap and let you know! 👍

strange! Thanks for taking a look into it!

> I have left a note on my description on all that. Selecting a Site, Clicking the Create a new site button and the Landscape UI will be done in individual PR, separate to this one, you can also check the list of todos in the parent issue, here. Also, I just added the Jetpack App todo item, which will be done in a separate PR as well. Divide and conquer! 😄

Nice! Sorry I missed that. I'll have a look at the parent task!

> Definitely, let's discuss our options together. In this instance I used the @style/Widget.LoginFlow.Button.Secondary style that is already the style for the secondary (white) button during the login flow, as it is already defined by the library. I didn't want to make any changes to that before you can see the result and suggest updates to the button. 👍

Is that the button you're referring to? I was talking about the subheading. We can discuss on our call in a bit 😄

File: WordPress/src/main/res/layout/login_epilogue_screen_new.xml
Comment (osullivanchris) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: @ParaskP7 that sounds fine. Good to be consistent and stick to the 8dp grid. But I can take a look at a build/screenshot of both screens when its done just to be sure and give it a final 👍

**Comment by ParaskP7** [comment]
Thank YOU @osullivanchris !

> strange! Thanks for taking a look into it!

True, it was just our IDE messing with reordering XML elements automatically, which is causing that, and this is because of the predefined setup we are all using. We also talked with @ashiagr to extract this login in a separate layout and make it reusable. This will avoid further such problems in the future and make it easier for anyone else to reuse this component. 👍 

> Nice! Sorry I missed that. I'll have a look at the parent task!

The description was long, of course! 🌟 

> Is that the button you're referring to? I was talking about the subheading. We can discuss on our call in a bit 😄

Oh, you are right, I misread, this is not about the `button`, but rather the `subheading`. In any ways, I am open for any suggestions you might have for me! 👍

File: WordPress/src/main/res/layout/login_epilogue_screen_new.xml
Comment (ParaskP7) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: 👋 @osullivanchris !

Yes, exactly, true. I'll then ping you when we get ready with the final APK and I'll have the updated screenshots.

PS: Bear in mind that this might take sometime since we are having some issues with the `Login` library but `Platform9` is working on fixing them as we speak.

Comment (ParaskP7) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: 👋 @ashiagr !

I have updated the PR with the extracted `login-or-layout` that is not being reused, could you please do another review on that: fbc7089c325b5aefee0c579dd102d1c1665655f0 🙏 

👋 @osullivanchris !

I have update the PR description screenshots to the how the `Or Layout` is looking now. Can you also please run through the app and verify that everything works as expected, including looking at the `login-email-screen` with the `Google Sign-In` button and its `Or Layout` and verify that it looks good as well? 🙏

Comment (ashiagr) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: 👋 @ParaskP7,

I've reviewed fbc7089, also tested all screens including this layout, all LGTM 👍. 
I've approved the PR, feel free to merge once Chris approves it.

**Review by ashiagr — APPROVED** [review]

File: WordPress/src/main/res/layout/login_epilogue_screen_new.xml
Comment (ParaskP7) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: Thank you @ashiagr ! 🙏

**Comment by ParaskP7** [comment]
👋 @osullivanchris !

> Oh, you are right, I misread, this is not about the button, but rather the subheading. In any ways, I am open for any suggestions you might have for me! 👍

As discussed, I have just pushed a fix which updates the `textAppearanceHeadline6` text appearance to `textAppearanceSubtitle1`. I have also updated the description with the new screenshots. Please do let me know if everything looks ok now and thank you a lot for all your support. 🙏

**Comment by osullivanchris** [comment]
Looks good @ParaskP7 and thanks for the chat! I'm updating zeplin to reflect the same change 😄

File: WordPress/src/main/res/layout/login_epilogue_screen_new.xml
Comment (osullivanchris) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: > I have update the PR description screenshots to the how the Or Layout is looking now. Can you also please run through the app and verify that everything works as expected, including looking at the login-email-screen with the Google Sign-In button and its Or Layout and verify that it looks good as well? 🙏

@ParaskP7 you may have finally found something that is beyond my eye to spot 🤓  I can't tell, so its totally fine to go with this change if it improves the reusability of this pattern.

Comment (ParaskP7) [reply WordPress/src/main/res/layout/login_epilogue_screen_new.xml]: 👋 @osullivanchris !

> @ParaskP7 you may have finally found something that is beyond my eye to spot 🤓 I can't tell, so its totally fine to go with this change if it improves the reusability of this pattern.

Nice, thanks so much for double checking this Chris, and for the 👍 ! We are now good to 🚀

**Review by osullivanchris — APPROVED** [review]

**Pull request merged**
